modules/order/src/Element/ProfileSelect.php

```diff
@@ -76,30 +102,115 @@ public function getInfo() {
-      throw new \InvalidArgumentException('The commerce_profile_select #default_value property must be a profile entity.');
-    }
     if (!is_array($element['#available_countries'])) {
       throw new \InvalidArgumentException('The commerce_profile_select #available_countries property must be an array.');
     }
+    if (empty($element['#profile_type'])) {
+      throw new \InvalidArgumentException('The commerce_profile_select #profile_type property must be provided.');
+    }
+    $entity_type_manager = \Drupal::entityTypeManager();
+    /** @var \Drupal\profile\ProfileStorageInterface $profile_storage */
+    $profile_storage = $entity_type_manager->getStorage('profile');
+    /** @var \Drupal\profile\Entity\ProfileTypeInterface $profile_type */
+    $profile_type = $entity_type_manager->getStorage('profile_type')->load($element['#profile_type']);
+
+    $user_profiles = [];
+    /** @var \Drupal\user\UserInterface $user */
+    $user = $entity_type_manager->getStorage('user')->load($element['#owner_uid']);
+
+    if (!$user->isAnonymous()) {
+      // If the user exists, attempt to load other profiles for selection.
+      foreach ($profile_storage->loadMultipleByUser($user, $profile_type->id(), TRUE) as $existing_profile) {
+        $user_profiles[$existing_profile->id()] = $existing_profile->label();
 
-    $element['#profile'] = $element['#default_value'];
-    $form_display = EntityFormDisplay::collectRenderDisplay($element['#profile'], 'default');
-    $form_display->buildForm($element['#profile'], $element, $form_state);
-    if (!empty($element['address']['widget'][0])) {
-      $widget_element = &$element['address']['widget'][0];
-      // Remove the details wrapper from the address widget.
-      $widget_element['#type'] = 'container';
-      // Provide a default country.
-      if (!empty($element['#default_country']) && empty($widget_element['address']['#default_value']['country_code'])) {
-        $widget_element['address']['#default_value']['country_code'] = $element['#default_country'];
+        // If this is the first form build, set the element's value based on
+        // the user's default profile.
+        if (!$form_state->isProcessingInput() && $existing_profile->isDefault()) {
+          $element['#value'] = $existing_profile->id();
+        }
       }
-      // Limit the available countries.
-      if (!empty($element['#available_countries'])) {
-        $widget_element['address']['#available_countries'] = $element['#available_countries'];
+    }
+
+    $id_prefix = implode('-', $element['#parents']);
+    $wrapper_id = Html::getUniqueId($id_prefix . '-ajax-wrapper');
+    $element = [
+      '#tree' => TRUE,
+      '#prefix' => '<div id="' . $wrapper_id . '">',
+      '#suffix' => '</div>',
+      // Pass the id along to other methods.
+      '#wrapper_id' => $wrapper_id,
+      '#element_mode' => $form_state->get('element_mode') ?: 'view',
```

Comment:
If the multiple `ProfileSelect` form elements exist in the same form (eg. in checkout page, the shipping info and billing info), the `element_mode` in `$form_state` should be separated by the form parents. 

For example, commerce_shipping module is installed, and there are two form elements in the same checkout form page. Once a `edit_button` within shipping info (or payment info) is clicked, the same `element_mode` value in `$form_state` is set into _edit_ by `ajaxSubmit()`.  Therefore, `#element_mode` in shipping info and payment info are _edit_. It caused another form element in payment info doesn't contain `edit_button`. 

Perhaps, the element mode name should be with the form parents.

```
$form_state->get('element_mode-' . $id_prefix . '-edit_button')
```